src/NServiceBus.AcceptanceTests/Diagnostics/When_sending_messages_from_pipeline.cs

```diff
@@ -0,0 +1,69 @@
+﻿namespace NServiceBus.AcceptanceTests.Diagnostics;
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using AcceptanceTesting;
+using EndpointTemplates;
+using NUnit.Framework;
+
+[NonParallelizable] // Ensure only activities for the current test are captured
+public class When_sending_messages_from_pipeline : NServiceBusAcceptanceTest
+{
+    [Test]
+    public async Task Should_add_batch_dispatch_events()
+    {
+        using var activityListener = TestingActivityListener.SetupNServiceBusDiagnosticListener();
+        var context = await Scenario.Define<Context>()
+            .WithEndpoint<TestEndpoint>(b => b
+                .When(s => s.SendLocal(new TriggerMessage())))
+            .Done(c => c.OutgoingMessageReceived)
+            .Run();
+
+        Assert.AreEqual(activityListener.CompletedActivities.Count, activityListener.StartedActivities.Count, "all activities should be completed");
+
+        var outgoingMessageActivities = activityListener.CompletedActivities.GetIncomingActivities();
+        var sentMessage = outgoingMessageActivities.First();
+
+        Assert.IsNotEmpty(sentMessage.Events);
+        var startDispatchingEvent = sentMessage.Events.Single(e => e.Name == "Start dispatching");
+        Assert.NotNull(startDispatchingEvent, "should raise dispatch start event");
```

Comment:
I think that `.Single(...)` will throw if there's no matching elements anyway so this assert will never fail. Maybe change the above to SingleOrDefault so that the problem gets caught by the Assert for clarity.

Reply:
SingleOrDefault might not be the best idea since ActivityEvent is a struct and the default part would return an instance even when no event has been found. I changed not do redundant checks though.